src/client/eppo-client.ts

```diff
@@ -216,303 +181,208 @@ export default class EppoClient implements IEppoClient {
-    }
-  }
-
   public getStringAssignment(
     subjectKey: string,
     flagKey: string,
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    subjectAttributes: Record<string, any> = {},
-    assignmentHooks?: IAssignmentHooks | undefined,
+    defaultValue: string,
+    subjectAttributes: Record<string, AttributeType> = {},
     obfuscated = false,
-  ): string | null {
-    try {
-      return (
-        this.getAssignmentVariation(
-          subjectKey,
-          flagKey,
-          subjectAttributes,
-          assignmentHooks,
-          obfuscated,
-          ValueType.StringType,
-        ).stringValue ?? null
-      );
-    } catch (error) {
-      return this.rethrowIfNotGraceful(error);
-    }
+  ): string {
+    return (
+      this.getAssignmentVariation(
+        subjectKey,
+        flagKey,
+        EppoValue.String(defaultValue),
+        subjectAttributes,
+        obfuscated,
+        VariationType.STRING,
+      ).stringValue ?? defaultValue
+    );
   }
 
   getBoolAssignment(
     subjectKey: string,
     flagKey: string,
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    subjectAttributes: Record<string, any> = {},
-    assignmentHooks?: IAssignmentHooks | undefined,
+    defaultValue: boolean,
+    subjectAttributes: Record<string, AttributeType> = {},
     obfuscated = false,
-  ): boolean | null {
-    try {
-      return (
-        this.getAssignmentVariation(
-          subjectKey,
-          flagKey,
-          subjectAttributes,
-          assignmentHooks,
-          obfuscated,
-          ValueType.BoolType,
-        ).boolValue ?? null
-      );
-    } catch (error) {
-      return this.rethrowIfNotGraceful(error);
-    }
+  ): boolean {
+    return (
+      this.getAssignmentVariation(
+        subjectKey,
+        flagKey,
+        EppoValue.Bool(defaultValue),
+        subjectAttributes,
+        obfuscated,
+        VariationType.BOOLEAN,
+      ).boolValue ?? defaultValue
+    );
   }
 
   getNumericAssignment(
     subjectKey: string,
     flagKey: string,
-    subjectAttributes?: Record<string, EppoValue>,
-    assignmentHooks?: IAssignmentHooks | undefined,
+    defaultValue: number,
+    subjectAttributes?: Record<string, AttributeType>,
     obfuscated = false,
-  ): number | null {
-    try {
-      return (
-        this.getAssignmentVariation(
-          subjectKey,
-          flagKey,
-          subjectAttributes,
-          assignmentHooks,
-          obfuscated,
-          ValueType.NumericType,
-        ).numericValue ?? null
-      );
-    } catch (error) {
-      return this.rethrowIfNotGraceful(error);
-    }
+  ): number {
+    return (
+      this.getAssignmentVariation(
+        subjectKey,
+        flagKey,
+        EppoValue.Numeric(defaultValue),
+        subjectAttributes,
+        obfuscated,
+        VariationType.NUMERIC,
+      ).numericValue ?? defaultValue
+    );
   }
 
-  public getJSONStringAssignment(
+  getIntegerAssignment(
```

Comment:
🔥 

Minor, but consider listing this before `getNumericAssignment()` as its a more specific subset in a way

Reply:
Can switch them, but note it is not a more specific subset. They are mutually exclusive. Something is either numeric (a float), or an integer. You cannot use `getNumericAssignment()` on an integer flag